Ext function form (#1842)
kyma-project/busola: docs/extensibility/form-section.md

The **form** section contains a list of objects that define which fields are included in the final form. All given fields are placed in the advanced form by default. It's possible to add a field to the simple form by providing the `simple: true` flag. You can also remove it from the advanced form by providing the `advanced: false` flag.

> Any parameters that are not handled by the widget are added to the schema directly, so it's possible to add or override existing values. For example, add an **enum** parameter to provide selectable values in a field or specify additional parameters to improve the schema validation, e.g. `min` and `max` attributes for numeric inputs to enable HTML validation.

Reviewer: ```suggestion
Any parameters that are not handled by the widget are added to the schema directly, so it's possible to add or override existing values. For example, add an **enum** parameter to provide selectable values in a field or specify additional parameters to improve the schema validation, for example, `min` and `max` attributes for numeric inputs to enable HTML validation.
```

#### Widget-specific parameters

> - **language** - a jsonata expression resolving the desired language. It has access to `$root` variable, containing the entire resource.

Reviewer: ```suggestion
- **language** - a JSONata expression resolving the desired language. It has access to the `$root` variable, containing the entire resource.
```

- **inputInfo** - a string below the input field that shows how to fill in the input.
- **description** - a string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).

#### Example

```json
{
  "path": "spec.my-data",
  "widget": "CodeEditor",
  "path": "spec.data",
  "widget": "CodeEditor",
  "inputInfo": "Data needs to be a valid JSON object.",

>   "description": "Data that will be passed on to the application.",

Reviewer: ```suggestion
  "description": "Data is passed on to the application.",
```

  "language": "'JSON'"
}
```

<img src="./assets/form-widgets/CodeEditor.png" alt="Example of a code editor widget" style="border: 1px solid #D2D5D9">

> > NOTE: Remember to put both single and double quotes if you want to use plain language (e.g. "'YAML'"). Specyfing just double quotes ("YAML") makes Busola try to access nonexistent `YAML` variable, resulting the language to be `undefined`.

Reviewer: ```suggestion
> NOTE: Remember to put both single and double quotes if you want to use plain language (for example, "'YAML'"). Specifying just double quotes ("YAML") makes Busola try to access a nonexistent `YAML` variable, resulting in the language being `undefined`.
```

FormGroup widgets render an `object` as a collapsible section.

#### Widget-specific parameters

> - **columns** - number of columns the content should be rendered in. Defaults to 1.

Reviewer: ```suggestion
- **columns** - number of columns the content is rendered in. Defaults to 1.
```
